lib/publiccloud/instance.pm

```diff
@@ -399,21 +430,6 @@ sub wait_for_ssh {
     my $instance_msg = "instance: $self->{instance_id}, public IP: $self->{public_ip}";
     my ($duration, $exit_code, $sshout, $sysout);
 
-    # skip SLES4SAP as incompatible with get_public_ip
```

Comment:
To me it is ok the external call, but complex; so eventually, as I wrote [here](https://suse.slack.com/archives/C02CGKBCGT1/p1756380328813069?thread_ts=1756293271.534829&cid=C02CGKBCGT1), to simplify the maintenance, the new routine could be called **here** directly, mantaining its objects indipendency, but avoiding to add it before every `wait_for_ssh` call. Up to you 👍🏼